chef-utils/lib/chef-utils/internal.rb

```diff
@@ -85,6 +85,8 @@ def __transport_connection
       # Software consumers MUST override this method with their own implementation.  The default behavior here is subject to change.
       return Chef.run_context.transport_connection if defined?(Chef) && Chef.respond_to?(:run_context) && Chef&.run_context&.transport_connection
 
+      return Chef.bootstrap_context.transport_connection if defined?(Chef) && Chef.respond_to?(:bootstrap_context) && Chef&.bootstrap_context&.transport_connection
+
```

Comment:
- As run context denoting the chef-client run and have ohai node object in place to respond query node info like platform etc.

- Here we have added bootstrap_context which denoting we have actively made connection to the node able to answer query about the node using train transport connection.

Reply:
This likely isn't the right way to do this for knife.  Chef has to do this because we support calling the DSL methods from library files where we don't have any kind of useful context or DSL so need to use the Chef.run_context to get at the state of the chef-client run.  That itself is probably a bad idea because it isn't threadsafe.  What should be done here is that under lib/chef/knife there should be a mixin created which pulls in the train_helpers and wires up the transport_connection similarly to lib/ohai/mixin/train_helpers.rb in https://github.com/chef/ohai/pull/1503/files